pkg/namesgenerator/names-generator.go

```diff
@@ -840,13 +841,31 @@ var (
 // integer between 0 and 10 will be added to the end of the name, e.g `focused_turing3`
 func GetRandomName(retry int) string {
 begin:
-	name := fmt.Sprintf("%s_%s", left[rand.Intn(len(left))], right[rand.Intn(len(right))]) //nolint:gosec // G404: Use of weak random number generator (math/rand instead of crypto/rand)
+	// Generating a random number for the left part of the name.
+	l, err := rand.Int(rand.Reader, big.NewInt(int64(len(left))))
+	if err != nil {
+		goto begin
+	}
+	// Generating a random number for the right part of the name.
+	r, err := rand.Int(rand.Reader, big.NewInt(int64(len(right))))
+	if err != nil {
+		goto begin
+	}
+
+	name := fmt.Sprintf("%s_%s", left[l.Int64()], right[r.Int64()]) //nolint:gosec // G404: Use of weak random number generator (math/rand instead of crypto/rand)
```

Comment:
Looks like we no longer need the `//nolint` lines if this was changed. 

On a side-not; we should remove the use of sprintf for these, and instead just concatenate;

```suggestion
	name := left[l.Int64()] + "_" + right[r.Int64()]
```

Reply:
About the using of sprintf we had no choice, because we are 2 integers and one string. Maybe we can use strconv instead of sprintf?